### PR TITLE
User controller

### PR DESCRIPTION
**Pull request by praveendraa**

### CODE REVIEW

File: TaskManagement-WebService/WebApplication1/DTO/UserUpdateDTO.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿namespace WebApplication1.DTO
+{    public class UserUpdateDTO
+    {
+        public string Name { get; set; } // Optional
```
Comment (dukpapp) [inline TaskManagement-WebService/WebApplication1/DTO/UserUpdateDTO.cs]: Can you take the comment out from the code?

File: TaskManagement-WebService/WebApplication1/DTO/UserDTO.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿namespace WebApplication1.DTO
+{
+    public class UserDTO
+    {
+        public string ID { get; set; }
+        public string Name { get; set; }
+        public string Title { get; set; }
+        public string Bio { get; set; }
+        public DateTime CreatedDate { get; set; } = DateTime.Now;
```
Comment (dukpapp) [inline TaskManagement-WebService/WebApplication1/DTO/UserDTO.cs:9]: I like this!

File: TaskManagement-WebService/WebApplication1/Controllers/UserController.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+        public IActionResult UserUpdate (string Name, string Bio, string Title)
+        {
+           var update_users = new List<UserUpdateDTO>();
+           return Ok(update_users);
+        }
+
+        //DELETE /api/users/{userId}: Delete user.
+        [HttpDelete("users")]
+        public IActionResult DeleteUser()
```
Comment (dukpapp) [inline TaskManagement-WebService/WebApplication1/Controllers/UserController.cs]: For now, can we delete the route. May be we can get this one added in the future.

File: TaskManagement-WebService/WebApplication1/Controllers/UserController.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            return Ok(users);
+        }
+
+        //GET /api/users/{userId} Get user details by userId.
+        [HttpGet("UserID")]
+        
+        public IActionResult GetUser(string ID)
+        {  
+            
+            return Ok();
```
Comment (dukpapp) [inline TaskManagement-WebService/WebApplication1/Controllers/UserController.cs]: Can you return empty User here? The main purpose of route is to understand the API contract.

File: TaskManagement-WebService/WebApplication1/Controllers/UserController.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        public IActionResult GetUser(string ID)
+        {  
+            
+            return Ok(); 
+        }
+
+        //PUT /api/users/{userId}Update user details.
+        [HttpPut("update_users")]
+
+        public IActionResult UserUpdate (string Name, string Bio, string Title)
```
Comment (dukpapp) [inline TaskManagement-WebService/WebApplication1/Controllers/UserController.cs]: How do I know which user to update without any unique identifier?

File: TaskManagement-WebService/WebApplication1/Controllers/UserController.cs [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        {
+           var update_users = new List<UserUpdateDTO>();
+           return Ok(update_users);
```
Comment (dukpapp) [inline TaskManagement-WebService/WebApplication1/Controllers/UserController.cs:42]: I don't think we can update multiple user. Always one user at a time

File: TaskManagement-WebService/WebApplication1/Controllers/UserController.cs (hunk 3 above)
Comment (praveendraa) [reply TaskManagement-WebService/WebApplication1/Controllers/UserController.cs]: Done

File: TaskManagement-WebService/WebApplication1/Controllers/UserController.cs (hunk 5 above)
Comment (praveendraa) [reply TaskManagement-WebService/WebApplication1/Controllers/UserController.cs]: Done

**Review by dukpapp — APPROVED** [review]
Grea Job! Merge whenever

**Pull request merged**